core/src/main/java/org/mule/runtime/core/el/BindingContextUtils.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.el;
+
+import static java.util.Collections.unmodifiableMap;
+import static java.util.Objects.requireNonNull;
+import static org.mule.runtime.api.metadata.DataType.STRING;
+import static org.mule.runtime.api.metadata.DataType.fromType;
+
+import org.mule.runtime.api.el.BindingContext;
+import org.mule.runtime.api.message.Error;
+import org.mule.runtime.api.message.Message;
+import org.mule.runtime.api.metadata.DataType;
+import org.mule.runtime.api.metadata.TypedValue;
+import org.mule.runtime.core.api.Event;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Provides a reusable way for creating {@link BindingContext}s.
+ * 
+ * @since 4.0
+ */
+public class BindingContextUtils {
+
+  public static final String PAYLOAD = "payload";
+  public static final String DATA_TYPE = "dataType";
+  public static final String ATTRIBUTES = "attributes";
+  public static final String ERROR = "error";
+  public static final String ID = "id";
+  public static final String CORRELATION_ID = "correlationId";
+  public static final String VARIABLES = "variables";
+  public static final String PROPERTIES = "properties";
+  public static final String PARAMETERS = "parameters";
+  public static final String FLOW = "flow";
+  public static final String SERVER = "server";
+  public static final String MULE = "mule";
+  public static final String APP = "app";
+
+  public static final BindingContext NULL_BINDING_CONTEXT = BindingContext.builder().build();
+
+  private BindingContextUtils() {
+    // Nothing to do
+  }
+
+  /**
+   * Creates a new {@link BindingContext} that contains the bindings from {@code baseContext} and the bindings that belong to the
+   * given {@code event}.
+   * 
+   * @param event the event to build the new bindings for. Not-null.
+   * @param baseContext the context whose copy the event bindings will be added to. Not-null.
+   * @return a new {@link BindingContext} that contains the bindings from {@code baseContext} and the bindings that belong to the
+   *         given {@code event}.
+   */
+  public static BindingContext addEventBindings(Event event, BindingContext baseContext) {
+    requireNonNull(event);
+    requireNonNull(baseContext);
+
+    BindingContext.Builder contextBuilder = BindingContext.builder(baseContext);
+
+    contextBuilder.addBinding(ID, new TypedValue<>(event.getContext().getId(), STRING));
+    contextBuilder.addBinding(CORRELATION_ID, new TypedValue<>(event.getContext().getCorrelationId(), STRING));
+    Map<String, TypedValue> flowVars = new HashMap<>();
```

Comment:
If I remember correctly there's an issue regarding what to do when the user tries to set a variable with the same name as ours, until that is done we should keep the behaviour of setting the flow vars first and then all of our bindings, so they are overridden. Can you fix that for ID and CORRELATION ID?